_query-dsl/specialized/script.md

```diff
@@ -0,0 +1,215 @@
+---
+layout: default
+title: Script query
+parent: Specialized queries
+nav_order: 58
+---
+
+# Script query
+
+Use the `script` query to filter documents based on a custom condition written in the Painless scripting language. This query returns documents for which the script evaluates to `true`, enabling advanced filtering logic that can't be expressed using standard queries.
+
+The `script` query is computationally expensive and should be used sparingly. Only use it when necessary and ensure `search.allow_expensive_queries` is enabled (default is `true`).
+
+## Parameters
+
+| Parameter       | Required/Optional | Description                                           |
+| --------------- | ----------------- | ----------------------------------------------------- |
+| `script.source` | Required          | The script code that evaluates to `true` or `false`.  |
+| `script.params` | Optional          | User-defined parameters referenced inside the script. |
+
+## Create a sample index
+
+```json
+PUT /products
+{
+  "mappings": {
+    "properties": {
+      "title": { "type": "text" },
+      "price": { "type": "float" },
+      "rating": { "type": "float" }
+    }
+  }
+}
+```
+
+{% include copy-curl.html %}
+
+## Index example documents
+
+```json
+POST /products/_bulk
+{ "index": { "_id": 1 } }
+{ "title": "Wireless Earbuds", "price": 99.99, "rating": 4.5 }
+{ "index": { "_id": 2 } }
+{ "title": "Bluetooth Speaker", "price": 79.99, "rating": 4.8 }
+{ "index": { "_id": 3 } }
+{ "title": "Noise Cancelling Headphones", "price": 199.99, "rating": 4.7 }
+```
+
+{% include copy-curl.html %}
+
+## Basic script query
+
+Return products with a rating above `4.6`:
+
+```json
+POST /products/_search
+{
+  "query": {
+    "script": {
+      "script": {
+        "source": "doc['rating'].value > 4.6"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The returned hits only include documents with `rating` higher than `4.6`:
+
+```json
+{
+  ...
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": "products",
+        "_id": "2",
+        "_score": 1,
+        "_source": {
+          "title": "Bluetooth Speaker",
+          "price": 79.99,
+          "rating": 4.8
+        }
+      },
+      {
+        "_index": "products",
+        "_id": "3",
+        "_score": 1,
+        "_source": {
+          "title": "Noise Cancelling Headphones",
+          "price": 199.99,
+          "rating": 4.7
+        }
+      }
+    ]
+  }
+}
+```
+
+## Use `params` in script
```

Comment:
This section can stay after the Parameters section. Let's call it "Using script parameters"